File: client/java/src/com/tibco/tgdb/model/impl/GraphManagerImpl.java
/**
 * Copyright 2016 TIBCO Software Inc. All rights reserved.
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); You may not use this file except 
 * in compliance with the License.
 * A copy of the License is included in the distribution package with this file.
 * You also may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * <p/>
 * File name : GraphManager.${EXT}
 * Created on: 1/23/15
 * Created by: suresh 
 * <p/>
 * SVN Id: $Id: GraphManagerImpl.java 622 2016-03-19 20:51:12Z ssubrama $
 */


package com.tibco.tgdb.model.impl;

import com.tibco.tgdb.exception.TGException;
import com.tibco.tgdb.model.*;
import com.tibco.tgdb.query.TGFilter;
import com.tibco.tgdb.query.TGQuery;
import com.tibco.tgdb.query.TGResultSet;
import com.tibco.tgdb.query.TGTraversalDescriptor;

import java.util.Set;

public class GraphManagerImpl implements TGGraphManager {

    @Override
    public TGNode createNode() throws TGException {
        return null;
    }

    @Override
    public TGNode createNode(TGNodeType nodeType) throws TGException {
        return null;
    }

    @Override
    public TGEdge createEdge(TGNode fromNode, TGNode toNode, TGEdge.DirectionType directionType) throws TGException {
        return null;
    }

    @Override
    public TGEdge createEdge(TGNode fromNode, TGNode toNode, TGEdgeType edgeType) throws TGException {
        return null;
    }

    @Override
    public TGGraph createGraph(String name) throws TGException {
        return null;
    }

    @Override
    public TGResultSet queryNodes(TGFilter filter, Object... args) {
        return null;
    }

    @Override
    public TGQuery createQuery(TGFilter filter) {
        return null;
    }

    @Override
    public TGResultSet traverse(TGTraversalDescriptor descriptor, Set<TGNode> startingPoints) {
        return null;
    }

    @Override
    public void deleteNode(TGFilter filter) throws Exception {

    }

    @Override
    public void deleteNodes(TGFilter filter) throws Exception {

    }

    @Override
    public TGGraphMetadata getGraphMetadata() {
        return null;
    }
}

File: client/goAPI/model/GraphManagerImpl.go
package model

import (
	"github.com/TIBCOSoftware/tgdb-client/client/goAPI/types"
)

/**
 * Copyright 2018-19 TIBCO Software Inc. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); You may not use this file except
 * in compliance with the License.
 * A copy of the License is included in the distribution package with this file.
 * You also may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF DirectionAny KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * File name: TGGraphManager.go
 * Created on: Oct 06, 2018
 * Created by: achavan
 * SVN id: $id: $
 *
 */

type GraphManager struct {
	name string
}

func NewGraphManager(gmd GraphMetadata) GraphManager {
	newGraphManager := GraphManager{
		name: "TGDB Graph Manager",
	}
	return newGraphManager
}

///////////////////////////////////////
// Helper functions for GraphManager //
///////////////////////////////////////

// GetName gets Graph Manager's name
func (obj *GraphManager) GetName() string {
	return obj.name
}

///////////////////////////////////////////////////////////
// Implement functions from Interface ==> TGGraphManager //
///////////////////////////////////////////////////////////

// CreateNode creates Node within this Graph. There is a default Root Graph.
func (obj *GraphManager) CreateNode() (types.TGNode, types.TGError) {
	return nil, nil
}

// CreateNodeForNodeType creates Node of particular Type
func (obj *GraphManager) CreateNodeForNodeType(nodeType types.TGNodeType) (types.TGNode, types.TGError) {
	return nil, nil
}

// CreateEdge creates an Edge
func (obj *GraphManager) CreateEdge(fromNode types.TGNode, toNode types.TGNode, edgeType int) (types.TGEdge, types.TGError) {
	return nil, nil
}

// CreateEdgeWithDirection creates an Edge with direction
func (obj *GraphManager) CreateEdgeWithDirection(fromNode types.TGNode, toNode types.TGNode, directionType types.TGDirectionType) (types.TGEdge, types.TGError) {
	return nil, nil
}

// CreateGraph creates a SubGraph at the Root level.
func (obj *GraphManager) CreateGraph(name string) (types.TGGraph, types.TGError) {
	return nil, nil
}

// DeleteNode removes this node from the graph
func (obj *GraphManager) DeleteNode(filter types.TGFilter) (types.TGGraphManager, types.TGError) {
	return nil, nil
}

// DeleteNodes removes the nodes from this graph that match the filter
func (obj *GraphManager) DeleteNodes(filter types.TGFilter) (types.TGGraphManager, types.TGError) {
	return nil, nil
}

// CreateQuery creates a Reusable Query
func (obj *GraphManager) CreateQuery(filter types.TGFilter) types.TGQuery {
	return nil
}

// QueryNodes gets Nodes based on the Filter condition with a set of Arguments
func (obj *GraphManager) QueryNodes(filter types.TGFilter, args ...interface{}) types.TGResultSet {
	return nil
}

// Traverse follows the graph using the traversal descriptor
func (obj *GraphManager) Traverse(descriptor types.TGTraversalDescriptor, startingPoints []types.TGNode) types.TGResultSet {
	return nil
}

// GetGraphMetadata gets the Graph Metadata
func (obj *GraphManager) GetGraphMetadata() types.TGGraphMetadata {
	return nil
}
